src/AFQMC/Wavefunctions/WavefunctionFactory.cpp

```diff
@@ -734,15 +736,14 @@ Wavefunction WavefunctionFactory::fromHDF5(TaskGroup_& TGprop,
   hdf_archive restart(TGwfn.Global());
   if (restart_file != "")
   {
-    if (type == "phmsd")
-    {
-      app_log() << " Restarting from PHMSD trial wavefunction not implemented. \n";
-    }
-    else
-    {
+    app_log() << " Open " << restart_file << " for HamOps\n";
+    // first, try open existing restart file
+    if (!restart.open(restart_file, H5F_ACC_RDONLY))
+    { // make new restart file if cannot open existing
+      app_log() << " No restart_file create anew.\n";
       if (!restart.create(restart_file, H5F_ACC_EXCL))
       {
-        app_error() << " Error opening restart file in WavefunctionFactory. \n";
+        app_error() << " Error in WavefunctionFactory: Failed to create restart_file" << restart_file << "\n";
```

Comment:
A separate issue from this PR.
If AFQMC error handling is the same as real-space part, then this error handling is probably wrong in the case of MPI runs unless this function is only called from rank 0. If all the MPI ranks hit here, app_error goes out on rank 0 only but other ranks hit abort first, then the error message is not visible to the user.

Reply:
Good point. Try to use the modern abort procedures consistently.